ai_diffusion/workflow.py

```diff
@@ -680,6 +681,14 @@ def scale_refine_and_decode(
         assert mode is ScaleMode.upscale_quality
         upscaler = models.upscale[UpscalerName.default]
 
+    # if an canvas deviates both sizes from 1024 huge performance penalty tiled vae decreases it this is intel only
+    if (
+        extent.desired.width > 1536
+        or extent.desired.height > 1536
+        and Server.backend is ServerBackend.xpu
```

Comment:
`Server.backend` is a class field, as far as I know it remains constant (only the instance field is changed)

Using `settings.server_backend` should work. Generally should avoid accessing globals when building workflows, but since it's a local performance optimization it should work.